src/ImageSharp/Processing/Processors/Transforms/AffineTransformProcessor.cs

```diff
@@ -20,29 +16,35 @@ namespace SixLabors.ImageSharp.Processing.Processors.Transforms
     /// Provides the base methods to perform affine transforms on an image.
     /// </summary>
     /// <typeparam name="TPixel">The pixel format.</typeparam>
-    internal class AffineTransformProcessor<TPixel> : InterpolatedTransformProcessorBase<TPixel>
+    internal class AffineTransformProcessor<TPixel> : TransformProcessorBase<TPixel>
         where TPixel : struct, IPixel<TPixel>
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="AffineTransformProcessor{TPixel}"/> class.
         /// </summary>
-        /// <param name="matrix">The transform matrix</param>
+        /// <param name="matrix">The transform matrix.</param>
         /// <param name="sampler">The sampler to perform the transform operation.</param>
-        /// <param name="targetDimensions">The target dimensions to constrain the transformed image to.</param>
-        public AffineTransformProcessor(Matrix3x2 matrix, IResampler sampler, Size targetDimensions)
-            : base(sampler)
+        /// <param name="sourceSize">The source image size.</param>
+        public AffineTransformProcessor(Matrix3x2 matrix, IResampler sampler, Size sourceSize)
         {
+            Guard.NotNull(sampler, nameof(sampler));
+            this.Sampler = sampler;
             this.TransformMatrix = matrix;
-            this.TargetDimensions = targetDimensions;
+            this.TargetDimensions = TransformUtils.GetTransformedSize(sourceSize, matrix);
```

Comment:
Not sure whether the logic implemented in `GetTransformedSize(...)` fits the needs of all users. Maybe we should expose a `AffineTransformProcessor(matrix, sampler, targetSize)` constructor instead, and deal with the rest in the extension methods.

Reply:
It’s seems to cover everything I could throw at it, shrinking and growing when need be. There’s actually similar code for handling free transforms in the net framework.